src/screens/track-screen/TrackScreenActionButtons.tsx

```diff
@@ -0,0 +1,212 @@
+  isOwner: boolean
+  isPublished?: boolean
+  isUnlisted: boolean
+  showFavorite: boolean
+  showOverflow: boolean
+  showRepost: boolean
+  showShare: boolean
+  trackId: ID
+}
+
+const createStyles = (themeColors: ThemeColors) =>
+  StyleSheet.create({
+    root: {
+      ...flexRowCentered(),
+      width: '100%',
+      flexDirection: 'row',
+      justifyContent: 'center',
+      borderBottomWidth: 1,
+      borderBottomColor: themeColors.neutralLight7,
+      height: 60,
+      paddingTop: 12,
+      paddingBottom: 8
+    },
+
+    actionButton: {
+      ...flexRowCentered(),
+      width: 30,
+      height: '100%',
+      justifyContent: 'center',
+      position: 'relative',
+      bottom: 1,
+      marginHorizontal: 16
+    },
+
+    icon: {
+      height: 30,
+      width: 30
+    },
+
+    iconWrapper: {
+      ...flexRowCentered(),
+      width: 30,
+      height: '100%',
+      overflow: 'visible',
+      marginHorizontal: 12
+    }
+  })
+
+/**
+ * The action buttons on track and playlist screens
+ */
+export const TrackScreenActionButtons = ({
+  hasReposted,
+  hasSaved,
+  isFollowing,
+  isOwner,
+  isPublished = true,
+  isUnlisted,
+  showFavorite,
+  showOverflow,
+  showRepost,
+  showShare,
+  trackId
+}: TrackScreenActionButtonsProps) => {
+  const styles = useThemedStyles(createStyles)
+  const { neutralLight4, neutralLight8 } = useThemeColors()
+  const dispatchWeb = useDispatchWeb()
+
+  const onToggleSave = () => {
+    if (!isOwner) {
+      if (hasSaved) {
+        dispatchWeb(unsaveTrack(trackId, FavoriteSource.TRACK_PAGE))
+      } else {
+        dispatchWeb(saveTrack(trackId, FavoriteSource.TRACK_PAGE))
+      }
+    }
+  }
+
+  const onToggleRepost = () => {
+    if (!isOwner) {
+      if (hasReposted) {
+        dispatchWeb(undoRepostTrack(trackId, RepostSource.TRACK_PAGE))
+      } else {
+        dispatchWeb(repostTrack(trackId, RepostSource.TRACK_PAGE))
+      }
+    }
+  }
+
+  const onShare = () => {
+    dispatchWeb(
+      requestOpenShareModal({
+        type: 'track',
+        trackId,
+        source: ShareSource.PAGE
+      })
+    )
+  }
+  const onPressOverflow = () => {
+    const overflowActions = [
+      isOwner || isUnlisted
+        ? null
+        : hasReposted
+        ? OverflowAction.UNREPOST
+        : OverflowAction.REPOST,
+      isOwner || isUnlisted
+        ? null
+        : hasSaved
+        ? OverflowAction.UNFAVORITE
+        : OverflowAction.FAVORITE,
+      OverflowAction.ADD_TO_PLAYLIST,
+      isFollowing
+        ? OverflowAction.UNFOLLOW_ARTIST
+        : OverflowAction.FOLLOW_ARTIST,
+      OverflowAction.VIEW_ARTIST_PAGE
+    ].filter(Boolean) as OverflowAction[]
+
+    dispatchWeb(
+      openOverflowMenu({
+        source: OverflowSource.TRACKS,
+        id: trackId,
+        overflowActions
+      })
+    )
+  }
+
+  const repostButton = (
+    <RepostButton
+      wrapperStyle={styles.actionButton}
+      onPress={onToggleRepost ?? (() => {})}
```

Comment:
might be nice to make these optional props?

Reply:
These didn't even need a null check because they are defined in this component, just a copy/paste remnant